### PR TITLE
capi: fork validator

### LINKED ISSUE #2058: Exposing C API for tip block execution

### PR DESCRIPTION
**Pull request by JacekGlen**
Closes #2058

Exposes Fork Validator endpoints to CAPI:
- verify chain
- fork choice update

Linked PR: https://github.com/erigontech/silkworm-go/pull/10

### CODE REVIEW

**Review by canepat — CHANGES_REQUESTED** [review]
First of all, you should integrate and test block execution at the chain tip within Erigon2, so you don't want to update `mdbx-go` version now.

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkworm/capi/silkworm.h [hunk 1]
```diff
@@ -248,6 +248,69 @@ SILKWORM_EXPORT int silkworm_sentry_start(SilkwormHandle handle, const struct Si
...
 SILKWORM_EXPORT int silkworm_sentry_stop(SilkwormHandle handle) SILKWORM_NOEXCEPT;
 
+
+//! Silkworm Fork Validator configuration options
+struct SilkwormForkValidatorSettings {
+    size_t batch_size;                    // {512 * 1024 * 1024};  // Batch size to use in stages
+    size_t etl_buffer_size;               // {256 * 1024 * 1024};  // Buffer size for ETL operations
+    uint32_t sync_loop_throttle_seconds;  // {0};                  // Minimum interval amongst sync cycle
+    bool stop_before_senders_stage;       // {false};              // Stop before senders stage
+};
```
[the remark this answers is not in the document]
Comment (JacekGlen) [reply silkworm/capi/silkworm.h:258]: Done

File: silkworm/capi/silkworm_test.cpp [hunk 2]
```diff
@@ -182,6 +186,37 @@ struct SilkwormLibrary {
...
+
+    int stop_fork_validator() const {
+        return silkworm_stop_fork_validator(handle_);
+    }
+
+    int fork_validator_verify_chain(silkworm::Hash head_hash) const {
+        bytes_32 head_hash_bytes{};
+        std::memcpy(head_hash_bytes.bytes, head_hash.bytes, 32);
+
+        auto result = new SilkwormForkValidatorValidationResult();
```
[the remark this answers is not in the document]
Comment (JacekGlen) [reply silkworm/capi/silkworm_test.cpp:201]: Done

File: silkworm/capi/silkworm_test.cpp [hunk 3]
```diff
@@ -182,6 +186,37 @@ struct SilkwormLibrary {
...
+
+        return silkworm_fork_validator_verify_chain(handle_, head_hash_bytes, result);
+
+        delete result;
```
[the remark this answers is not in the document]
Comment (JacekGlen) [reply silkworm/capi/silkworm_test.cpp:205]: Done

File: silkworm/capi/silkworm.h [hunk 4]
```diff
@@ -248,6 +248,69 @@ SILKWORM_EXPORT int silkworm_sentry_start(SilkwormHandle handle, const struct Si
...
+
+typedef struct bytes_32 {
+    /** The 32 bytes. */
+    uint8_t bytes[32];
+} bytes_32;
```
[the remark this answers is not in the document]
Comment (JacekGlen) [reply silkworm/capi/silkworm.h:263]: Done

**Review by canepat — APPROVED** [review]

**Pull request merged**